lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1490,6 +1490,21 @@ bool QuakeBridgeVisitor::VisitCallExpr(clang::CallExpr *x) {
           builder.create<quake::DiscriminateOp>(loc, resTy, measure));
     }
 
+    // We may have qubit values that come from struct members, in this
+    // case the value may be provided as a pointer, need to load it here.
```

Comment:
I don't think we really want classical pointers to quantum references. (We always get rid of the C++ references.)

Reply:
When you extract a member from a struct you use ComputePtrOp, which returns a pointer to the member. How do we get around that?

Reply:
I was overly terse.

This is a huge, huge change. Lots of stuff in the compiler is predicated on the fact that we use `quake.alloca` to allocate qubits. Those qubits are allocated and used and have either memory-reference semantics or value semantics, etc. Everything is structured so we can do inlining, loop unrolling, qubit factoring, scheduling, etc. to eventually end up with a small(ish) cardinal set of qubits and a DAG (quantum circuit).

Stashing qubit references (which would semantically be storing values with quantum memory semantics in variables with classical memory semantics) will break the previous paragraph. Instead of linear chains of def-use on qubits that can be discriminated from one another, this would mean qubits and quantum memory references can alias, at a minimum. If one could bound the number of qubits (total $n$) to $k$ but not their identities in a region of code, this could mean one could have something like $n \choose k$ path replications to create the DAG in the worst case.

Reply:
I will remove this and convert structs with quantum members to value semantics (insert / extract value)